R/modspec.R

```diff
@@ -865,9 +867,28 @@ autodec_find <- function(code) {
   if(length(code)==0) {
     return(NULL)
   }
-  ans <- regmatches(code, regexpr("[._[:alnum:]]+ *=([^=]|$)", code, perl = TRUE))
-  ans <- unique(sub(" *=.?$", "", ans, perl = TRUE))
-  ans[!grepl(".", ans, fixed = TRUE)]
+  m <- regexpr("[._[:alnum:]]+ *=([^=]|$)", code, perl = TRUE) 
```

Comment:
Narrow the search for autodec.

Reply:
This saves the match locations so we can look at what happens from the start of the line to the variable name.